core/server/master/src/main/java/alluxio/master/metastore/rocks/RocksBlockStore.java

```diff
@@ -0,0 +1,233 @@
+    initDb(mBaseDir);
+  }
+
+  @Override
+  public Optional<BlockMeta> getBlock(long id) {
+    byte[] meta;
+    try {
+      meta = mDb.get(mBlockMetaColumn, Longs.toByteArray(id));
+    } catch (RocksDBException e) {
+      throw new RuntimeException(e);
+    }
+    if (meta == null) {
+      return Optional.empty();
+    }
+    try {
+      return Optional.of(BlockMeta.parseFrom(meta));
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public void putBlock(long id, BlockMeta meta) {
+    try {
+      // Overwrites the key if it already exists.
+      mDb.put(mBlockMetaColumn, Longs.toByteArray(id), meta.toByteArray());
+    } catch (RocksDBException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public void removeBlock(long id) {
+    try {
+      mDb.delete(mBlockMetaColumn, Longs.toByteArray(id));
+    } catch (RocksDBException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public void clear() {
+    try {
+      initDb(mBaseDir);
+    } catch (RocksDBException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public List<BlockLocation> getLocations(long id) {
+    RocksIterator iter =
+        mDb.newIterator(mBlockLocationsColumn, new ReadOptions().setPrefixSameAsStart(true));
+    iter.seek(Longs.toByteArray(id));
+    List<BlockLocation> locations = new ArrayList<>();
+    for (; iter.isValid(); iter.next()) {
+      try {
+        locations.add(BlockLocation.parseFrom(iter.value()));
+      } catch (Exception e) {
+        throw new RuntimeException(e);
+      }
+    }
+    return locations;
+  }
+
+  @Override
+  public void addLocation(long id, BlockLocation location) {
+    byte[] key = toByteArray(id, location.getWorkerId());
+    try {
+      mDb.put(mBlockLocationsColumn, key, location.toByteArray());
+    } catch (RocksDBException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public void removeLocation(long blockId, long workerId) {
+    byte[] key = toByteArray(blockId, workerId);
+    try {
+      mDb.delete(mBlockLocationsColumn, key);
+    } catch (RocksDBException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public Iterator<Block> iterator() {
+    RocksIterator iter =
+        mDb.newIterator(mBlockMetaColumn, new ReadOptions().setPrefixSameAsStart(true));
+    iter.seekToFirst();
+    return new Iterator<Block>() {
+      @Override
+      public boolean hasNext() {
+        return iter.isValid();
+      }
+
+      @Override
+      public Block next() {
+        try {
+          return new Block(Longs.fromByteArray(iter.key()), BlockMeta.parseFrom(iter.value()));
+        } catch (Exception e) {
+          throw new RuntimeException(e);
+        } finally {
+          iter.next();
+        }
+      }
+    };
+  }
+
+  // Implementation based on
+  // https://google.github.io/guava/releases/21.0/api/docs/src-html/com/google/common/primitives/
```

Comment:
should we just upgrade Guava anyway ?

Reply:
The referenced method writes 1 long to a byte array. But we need to write 2 longs to a byte array, so we need to adapt their code.